sormas-ui/src/main/java/de/symeda/sormas/ui/events/EventController.java

```diff
@@ -1006,95 +1006,41 @@ public EventDto createNewEvent(Disease disease) {
 		return EventDto.build(FacadeProvider.getCountryFacade().getServerCountry(), UserProvider.getCurrent().getUser(), disease);
 	}
 
-	public void deleteAllSelectedItems(Collection<EventIndexDto> selectedRows, Runnable callback) {
+	public void deleteAllSelectedItems(Collection<EventIndexDto> selectedRows, EventGrid eventGrid) {
 
-		if (selectedRows.size() == 0) {
-			new Notification(
-				I18nProperties.getString(Strings.headingNoEventsSelected),
-				I18nProperties.getString(Strings.messageNoEventsSelected),
-				Type.WARNING_MESSAGE,
-				false).show(Page.getCurrent());
+		ControllerProvider.getDeleteRestoreController()
+			.deleteAllSelectedItems(
+				selectedRows,
+				getEligibleEvents(selectedRows),
+				getIneligibleEvents(selectedRows),
+				DeleteRestoreHandlers.forEvent(),
+				bulkOperationCallback(eventGrid, null));
+	}
+
+	public Collection<EventIndexDto> getEligibleEvents(Collection<EventIndexDto> selectedRows) {
```

Comment:
Please remove this and the `getIneligibleEvents` method and instead build those lists directly in `deleteAllSelectedItems`, because this produces performance overhead (e.g. the same method gets called twice unnecessarily). I.e. build the list of ineligible events there (using a stream instead of a for loop), store it in a variable and use the stored list to build the list of eligible events.